design/FY2018/AirSystemSizingAdjust_Std62_1_refactor.md

```diff
@@ -0,0 +1,97 @@
+
+SimAirServingZones.cc, modify routine InitAirLoops() -  move code from MixedAir.cc routine GetOAControllerInputs() into this routine to fill out AirLoopZoneInfo data structure earlier in the program flow.  This allows for using the data structure earlier in the system sizing calculations, before the air system equipment is setup.  This allows for cleaner code in Standard 62.1 calculations which more simply loop over all zones on an air loop and not use the confusing code constructs based on separate "Cooled Zones" and "Heating Zones." 
+
+SimAirServingZones.cc, modify routine SetUpSysSizingArrays() - Move Standard 62.1 variables out to DataSizing.hh.  Remove existing system population diversity calculations (which are not accurate), add call to new DetermineSystemPopulationDiversity() routine.  Add code to support new SysAirMinFlowRatWasAutoSized variable in system sizing data structure that stores input for if the central heating air flow ratio was set to autosize on input. Revise handling of VozClgByZone and VozHtgByZone terms so that they do not already include system population diversity factor, to properly follow definition of Voz in Standard 62.1. Remove summary table reporting, to be replaced by new cleaner code in new routine ManageSystemVentilationAdjustments in SizingManager.cc
+
+SimAirServingZones.cc, modify routine UpdateSysSizing() - Move Standard 62.1 variables out to DataSizing.hh. Store state of new booleans sysSizeCoolingDominant and sysSizeHeatingDominant in FinalSysSizing data structure to know how the air system main design flow was determined. Remove summary table reporting, to be replaced by new cleaner code in new routine ManageSystemVentilationAdjustments in SizingManager.cc 
+
+DataSizing.cc/.hh add data variables for Standard 62.1 calculations, move variables buried in SimAirServingZones.cc subroutines and put into DataSizing.cc where the variables persist longer and can be accessed. 
+
+SizingManager.cc, new routine ManageSystemSizingAdjustments() - This routine adjusts system sizing outcomes based on how the zone air terminals finish out their sizing. The zone models are executed to trigger their sizing routines. Then the air terminal units data structures are scanned to sum design flow rates. Every air terminal connected to a particular air loop is summed for: 1) minimum heating flow rate, 2) maximum heating flow rate, 3) maximum flow rate, and 4) store zone level flow information for Standard 62.1 calculations, Vpz, Vpz_min, Vdz, and Vdz_min for both cooling and heating. The summed values are used to "Adjust" the system sizing results.  The corrected values are used to autosize the central heating flow ratio, if set to autosize by the user.  
```

Comment:
I haven't looked at the new code yet, so this may become obvious there, but I'm wondering how this timing fits in if Sizing:System specifies sizing based on VentilationRequirement or a hard value for Flow/System?  "This routine adjusts system sizing outcomes" - does this mean that the current sequence of zone sizing, system sizing, then call the terminal units is still in place, except for the 62.1 calcs?  Then the system sizing is adjusted?  What if the 62.1 calcs require that a particular zone have a higher supply flow (min or max) to meet the zone OA requirement? How does this get communicated back to the terminal unit?